diff --git a/node_modules/zero-md/src/zero-md.js b/node_modules/zero-md/src/zero-md.js
index 699f3d1..72ca56a 100644
--- a/node_modules/zero-md/src/zero-md.js
+++ b/node_modules/zero-md/src/zero-md.js
@@ -102,12 +102,14 @@
     _getInputs() {
       return new Promise((resolve, reject) => {
         // First try reading from light DOM template
-        let tpl = this.querySelector('template') && this.querySelector('template').content.querySelector('xmp') || false;
-        if (tpl) { resolve(tpl.textContent); return; }
+        let xmp = this.querySelector('template') && this.querySelector('template').content.querySelector('xmp') || false;
+        let script = this.querySelector('script[type="text/markdown"]') || false;
+        if (xmp) {resolve(xmp.textContent);return;}
+        if (script) {resolve(dedent(script.textContent));return;}
         // Next try reading from `src` attribute
-        this._ajaxGet(this.src)
-          .then(data => resolve(data))
-          .catch(err => reject(err));
+        this._ajaxGet(this.src).
+        then(data => resolve(data)).
+        catch(err => reject(err));
       });
     }
 
@@ -197,4 +199,11 @@
         });
     }
   });
+
+  function dedent(str) {
+    str = str.replace(/^\n/, "");
+    let match = str.match(/^\s+/);
+    return match ? str.replace(new RegExp("^"+match[0], "gm"), "") : str;
+  }
+
 }(window, document));
